tests/test_utils/test_materialize_arbitary_lazy_module.py

```diff
@@ -0,0 +1,23 @@
+import torch
+from colossalai.utils.model.lazy_init_context import LazyInitContext
+from torchvision.models import resnet34
+
+
+def test_lazy_init():
+    ctx = LazyInitContext()
+    with ctx:
+        model = resnet34(num_classes=10)
+    for param in model.parameters():
+        assert param.is_meta
+    for buffer in model.buffers():
+        assert buffer.is_meta
+    for module in model.children():
+        ctx.lazy_init_parameters(module)
+        for param in module.parameters():
+            assert not param.is_meta
+        for buffer in module.buffers():
+            assert not buffer.is_meta
```

Comment:
This test should check whether the model is initialized to the same parameters with and without lazy init given the same seed as well.

Reply:
Ok, I will add  numerical check in this test.